ext/serializer.c

```diff
@@ -1433,6 +1433,14 @@ void ddtrace_serialize_span_to_array(ddtrace_span_data *span, zval *array) {
     add_assoc_long(el, "duration", span->duration);
 
     zend_array *meta = ddtrace_property_array(&span->property_meta);
+    zend_array *metrics = ddtrace_property_array(&span->property_metrics);
+
+    // Remap OTel's status code (metric, http.response.status_code) to DD's status code (meta, http.status_code
+    zval *http_response_status_code = zend_hash_str_find(metrics, ZEND_STRL("http.response.status_code"));
+    if (http_response_status_code) {
+        zend_hash_str_update(meta, ZEND_STRL("http.status_code"), http_response_status_code);
```

Comment:
```suggestion
        Z_TRY_ADDREF_P(http_response_status_code);
        zend_hash_str_update(meta, ZEND_STRL("http.status_code"), http_response_status_code);
```
refcount in case this is a string or such